widget/entry.go

```diff
@@ -481,9 +481,13 @@ func (e *Entry) Append(text string) {
 	e.propertyLock.Lock()
 	provider := e.textProvider()
 	provider.insertAt(provider.len(), text)
-	e.updateText(provider.String())
+	changed := e.updateText(provider.String())
 	e.propertyLock.Unlock()
 
+	e.Validate()
+	if changed && e.OnChanged != nil {
+		e.OnChanged(provider.String())
```

Comment:
You should avoid calling `provider.String()` multiple times. If you look at the code you will see that it allocates an entirely new string each time. You probably want to save the string to a variable.

Reply:
np, fixed in next commit